### PR TITLE
Update ReadMe For tree

### PR DESCRIPTION
**Pull request by CraigMacomber**
## Description

Tree readme was a bit out of date. Update at least the parts at the top.

## Reviewer Guidance

The review process is outlined on [this wiki page](https://github.com/microsoft/FluidFramework/wiki/PR-Guidelines#guidelines).

### CODE REVIEW

File: packages/dds/tree/README.md [hunk 1]
```diff
@@ -11,15 +11,10 @@ and some of the implications of those goals for developers working on this packa
...
-    -   Removed content is retained forever.
-    -   Accessing an `EditableField` object (from its parent, e.g., with `getField`) in a loop will leak unbounded memory.
--   All changes are atomized by the `visitDelta` function.
-    This means that, when inserting/removing/moving 2 contiguous nodes, the `visitDelta` function will call the `DeltaVisitor` twice (once for each node) instead of once for both nodes.
-    Change notification consumers that are downstream from the `DeltaVisitor` will therefore also see those changes as atomized.
+-   The persisted format is now stable: documents created with released versions 2.0.0 of greater of this package will be supported long term.
+-   All range changes are atomized.
+    This means that, when inserting/removing/moving multiple contiguous nodes the edit is split up into separate single node edits.
+    This can impact the merge behavior of these edits, as well as the performance of large array edits.
 -   Some documentation (such as this readme and [the roadmap](docs/roadmap.md)) are out of date.
```
Comment (nmsimons) [inline packages/dds/tree/README.md:18]: If this is correct, surely we should update this readme, and not say it is out of date?

File: packages/dds/tree/README.md [hunk 2]
```diff
@@ -11,15 +11,10 @@ and some of the implications of those goals for developers working on this packa
...
--   The persisted format is not stable and long term support for persisted format is not yet in effect.
-    This means clients running more recent versions of SharedTree may find themselves unable to load old document or may corrupt them when editing.
-    Current estimate for LTS is April 2024.
--   SharedTree currently has unbounded memory growth:
-    -   Removed content is retained forever.
-    -   Accessing an `EditableField` object (from its parent, e.g., with `getField`) in a loop will leak unbounded memory.
--   All changes are atomized by the `visitDelta` function.
-    This means that, when inserting/removing/moving 2 contiguous nodes, the `visitDelta` function will call the `DeltaVisitor` twice (once for each node) instead of once for both nodes.
-    Change notification consumers that are downstream from the `DeltaVisitor` will therefore also see those changes as atomized.
+-   The persisted format is now stable: documents created with released versions 2.0.0 of greater of this package will be supported long term.
```
Comment (nmsimons) [inline packages/dds/tree/README.md:14]:
```suggestion
-   The persisted format is stable such that documents created with released versions 2.0.0 or greater of this package are fully supported long term.
```

File: packages/dds/tree/README.md [hunk 3]
```diff
@@ -11,15 +11,10 @@ and some of the implications of those goals for developers working on this packa
 
 Notable consideration that early adopters should be wary of:
 
--   The persisted format is not stable and long term support for persisted format is not yet in effect.
-    This means clients running more recent versions of SharedTree may find themselves unable to load old document or may corrupt them when editing.
-    Current estimate for LTS is April 2024.
--   SharedTree currently has unbounded memory growth:
-    -   Removed content is retained forever.
```
Comment (yann-achard-MS) [inline packages/dds/tree/README.md:18]: Until [https://github.com/microsoft/FluidFramework/pull/21372/](https://github.com/microsoft/FluidFramework/pull/21372/) is merged, removed content will keep on being retained forever.

File: packages/dds/tree/README.md (hunk 1 above)
Comment (CraigMacomber) [reply packages/dds/tree/README.md:18]: there are plenty of out of date things further below.

File: packages/dds/tree/README.md (hunk 3 above)
Comment (CraigMacomber) [reply packages/dds/tree/README.md:18]: I have updated the text to mention this.

File: packages/dds/tree/README.md [hunk 4]
```diff
@@ -11,15 +11,12 @@ and some of the implications of those goals for developers working on this packa
...
-    This means clients running more recent versions of SharedTree may find themselves unable to load old document or may corrupt them when editing.
-    Current estimate for LTS is April 2024.
+-   The persisted format is stable such that documents created with released versions 2.0.0 or greater of this package are fully supported long term.
 -   SharedTree currently has unbounded memory growth:
-    -   Removed content is retained forever.
-    -   Accessing an `EditableField` object (from its parent, e.g., with `getField`) in a loop will leak unbounded memory.
--   All changes are atomized by the `visitDelta` function.
-    This means that, when inserting/removing/moving 2 contiguous nodes, the `visitDelta` function will call the `DeltaVisitor` twice (once for each node) instead of once for both nodes.
-    Change notification consumers that are downstream from the `DeltaVisitor` will therefore also see those changes as atomized.
+    -   Removed content is retained in memory for the duration of the session ([fix](https://github.com/microsoft/FluidFramework/pull/21372)).
```
Comment (yann-achard-MS) [inline packages/dds/tree/README.md:16]:
```suggestion
    -   Removed content is retained in memory and persisted in the document at rest ([fix](https://github.com/microsoft/FluidFramework/pull/21372)).
```

**Review by yann-achard-MS — APPROVED** [review]

**Pull request merged**